mobify/platform-scaffold
# [Integration manager] global demandware connector

This PR updates how we manage some of the global app data for the salesforce connector. This includes updating `initDemandwareSession` to only create new sessions if we don't have an auth token stored as a cookie.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1281
 **Linked PRs**: #437 

## Changes
- Added initApp command to intialize the demand ware session
- Updated the initDemandwareSession command to only create a new session if we need to, and return the required request headers
- Updated existing DW commands to accept headers from initDemandwareSession

## How to test-drive this PR
- Run `npm run dev`
- Preview [the demandware site](https://preview.mobify.com/?url=http%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Navigate through the homepage, PLP and PDP with the network tab open
- Confirm everything is working and that there's only one request for `/s/Sites-2017refresh-Site/dw/shop/v17_2/customers/auth` and `s/Sites-2017refresh-Site/dw/shop/v17_2/sessions`

## Improvements for another PR
- Demandware's API allows us to refresh auth tokens, we should think about when/where we might need to do this. 
- How can we check if the current session is still valid?
- Add currency checking to the app initialization section
- Add users's login state (we might want to wait until we've looked at the login/create account sections)
